File: Prosary/Protocols/RosaryEngine.swift
//
//  RosaryEngine.swift
//  Prosary
//
//  Prayer-flow business logic boundary for the Rosary. Implement `StubRosaryEngine`
//  (see Support/Stubs/StubRosaryEngine.swift) with your real rules.
//

import Foundation

protocol RosaryEngine {
  /// Builds the full, ordered sequence of prayer steps for a Rosary session.
  func buildSteps(for prayer: Prayer) -> [RosaryStep]
}

File: app/src/main/java/com/dkaluta/Prosary/engine/RosaryEngine.kt
package com.dkaluta.Prosary.engine

import com.dkaluta.Prosary.models.Prayer
import com.dkaluta.Prosary.models.RosaryStep

/** What the UI needs from the backend to turn a saved favorite into an actual, ordered prayer
 * session. This is the prayer-flow business logic boundary. */
interface RosaryEngine {
    /** Builds the full, ordered sequence of prayer steps for a Rosary session. */
    fun buildSteps(prayer: Prayer): List<RosaryStep>
}
